README.md

```diff
@@ -1,9 +1,19 @@
 # DistributedSSIDatabase
+
 Distributed database, complete with serializable snapshot isolation, replication, and failure recovery
 
-Instructions to run a simulation:
-- run `mvn clean package` to compile and generate the jar file (located at target/DistributedSSI-1.0-SNAPSHOT.jar)
-- run the simulation for an input file `input1.txt` using command `java -jar DistributedSSI-1.0-SNAPSHOT.jar input1.txt` from the appropriate folder
+## SSI
+Serializable Snapshot Isolation is a form of Multi-version Read Consistency protocol. This means that a Read-only transaction will read committed data as of the time of the start of the transaction.
+
+In Snapshot Isolation, for Read-Write transactions, any writes follow the first committer wins rule at commit time, i.e., a transaction will only commit if no data items it wrote to were committed to by some other concurrent transactions.
 
-Disclaimer: Transactions are required to be named in the format Ti (where i=1,2,3...). 
-Variables need to be named xj (where j=1,2,3...)
+To ensure a serializable execution, we maintain a Serialization graph to detect cycles. Edges in the graph are labelled as r-w, w-w, or w-r edges. Cycle detection looks for a cycle with 2 r-w edges in a row; if such a cycle is found, we abort.
```

Comment:
We should probably say "we abort the transaction that induced the cycle" instead of "we abort"